### ISSUE TITLE
Testing issues

### ISSUE THREAD

**Issue by minhnimble** [abdulrahim46/Nimble#11]
It is quite nice to see that you've included a few unit tests and UI tests 👍. I am wondering if we could increase the unit test coverage to cover more the layers of the application? One thing you can and should focus on is actually the viewModel layer, where it contains the important logic in our app.

About the API tests you wrote, it is disturbing to see that you actually call the actual endpoints and wait for the result instead of using mock responses. Please consider update your tests with mock data instead.

**Comment by abdulrahim46** [comment]
Yes, I try to cover some unit tests but yeah agree with you on ViewModel layer, need to cover ViewModel 100% for unit tests. So, I was thinking to implement different protocols for different apis that we will conform in ViewModels; that way we can also mock apiclient by writing a MockAPIClient and conforming it to same protocols that we used in ViewModels and then we can mock data through it. What do you think?

**Comment by minhnimble** [comment]
Sounds good to me. I would like to see the implementation 👍
